test/functional/test_framework/test_node.py

```diff
@@ -294,9 +296,18 @@ def wait_for_cookie_credentials(self):
         self._raise_assertion_error("Unable to retrieve cookie credentials after {}s".format(self.rpc_timeout))
 
     def generate(self, nblocks, maxtries=1000000):
-        self.log.debug("TestNode.generate() dispatches `generate` call to `generatetoaddress`")
-        return self.generatetoaddress(nblocks=nblocks, address=self.get_deterministic_priv_key().address, maxtries=maxtries)
-
+        # if wallet is not compiled, blocks are generated to an address or descriptor using an HD seed
+        if '-disablewallet' in self.extra_args:
+            xpub = "tpubDAXcJ7s7ZwicqjprRaEWdPoHKrCS215qxGYxpusRLLmJuT69ZSicuGdSfyvyKpvUNYBW1s2U3NSrT6vrCYB9e6nZUEvrqnwXPF8ArTCRXMY"
+            finger_print = "d34db33f"
+            keypath = "/44'/0'/0'/0/" + str(self.hdkeypathnonce)
+            desc = "pkh([" + finger_print + keypath + "]" + xpub+ "/1)" 
+            self.hdkeypathnonce += 1
+            return self.generatetodescriptor(nblocks, desc, maxtries)
+        else:
+            hd_address = self.getnewaddress()
```

Comment:
Once I generate to a new address, many tests that check whether the wallet addr number matches up start failing. I think this is because the getnewaddress is messing up the address count for some tests. But at the same time, we also don't want to generate to deterministic addresses either... I'm not sure what direction I should take here...